crates/transpiler/src/passes/disjoint_layout.rs

```diff
@@ -0,0 +1,545 @@
+                    let cmap = subgraph(
+                        &coupling_map,
+                        &component
+                            .physical_qubits
+                            .iter()
+                            .map(|x| NodeIndex::new(x.index()))
+                            .collect(),
+                    );
+                    func(component.sub_dag, &cmap)
+                })
+                .collect::<PyResult<Vec<_>>>(),
+        )
+        .transpose(),
+    }
+}
+
+pub fn distribute_components(dag: &mut DAGCircuit, target: &Target) -> PyResult<DisjointSplit> {
+    let coupling_map: CouplingMap = match build_coupling_map(target) {
+        Some(map) => map,
+        None => return Ok(DisjointSplit::NoneNeeded),
+    };
+    let cmap_components = connected_components(&coupling_map);
+    if cmap_components.len() == 1 {
+        if dag.num_qubits() > cmap_components[0].len() {
+            return Err(TranspilerError::new_err(concat!(
+                "A connected component of the DAGCircuit is too large for any of the connected ",
+                "components in the coupling map."
+            )));
+        }
+        return Ok(DisjointSplit::NoneNeeded);
+    }
+    let dag_components = separate_dag(dag)?;
+    let mapped_components = map_components(&dag_components, &cmap_components)?;
+    let out_component_pairs: Vec<(DAGCircuit, CouplingMap)> = mapped_components
+        .into_iter()
+        .enumerate()
+        .filter(|(_, dag_indices)| !dag_indices.is_empty())
+        .map(|(cmap_index, dag_indices)| {
+            let mut out_dag = dag_components[*dag_indices.first().unwrap()].clone();
+            for dag_index in &dag_indices[1..] {
+                let dag = &dag_components[*dag_index];
+                for qubit in dag.qubits().objects() {
+                    out_dag.add_qubit_unchecked(qubit.clone())?;
+                }
+                for clbit in dag.clbits().objects() {
+                    out_dag.add_clbit_unchecked(clbit.clone())?;
+                }
+                for qreg in dag.qregs() {
+                    out_dag.add_qreg(qreg.clone())?;
+                }
+                for creg in dag.cregs() {
+                    out_dag.add_creg(creg.clone())?;
+                }
+                out_dag.compose(
+                    dag,
+                    Some(dag.qubits().objects()),
+                    Some(dag.clbits().objects()),
+                    false,
+                )?;
+            }
+            let subgraph = subgraph(&coupling_map, &cmap_components[cmap_index]);
+            Ok((out_dag, subgraph))
+        })
+        .collect::<PyResult<Vec<_>>>()?;
+    if out_component_pairs.len() == 1 {
+        return Ok(DisjointSplit::TargetSubset(
+            out_component_pairs[0]
+                .1
+                .node_weights()
+                .map(|x| PhysicalQubit::new(x.index() as u32))
+                .collect(),
+        ));
+    }
```

Comment:
For the future: I think we want to do this check before we even split the DAG.  I feel like the structure should be that we split the `Target` first, and then only split the DAG if we have to.

Reply:
Oh that's simple to qdd, I can do it tomorrow it's just add an additional check before L152 where we separate the dag to check if the number of active qubits in the full dag is less than the largest component.

Reply:
Added in: https://github.com/Qiskit/qiskit/pull/14177/commits/f5b0d2b9c9f9638d985f0465dbfb9b1a9e7d29ae